sonar-plugin-api/src/main/java/org/sonar/api/batch/bootstrap/ProjectDefinition.java

```diff
@@ -183,12 +183,9 @@ public String getVersion() {
     return properties.get(CoreProperties.PROJECT_VERSION_PROPERTY);
   }
 
+  @CheckForNull
   public String getName() {
-    String name = properties.get(CoreProperties.PROJECT_NAME_PROPERTY);
-    if (StringUtils.isBlank(name)) {
-      name = "Unnamed - " + getKey();
-    }
-    return name;
+    return properties.get(CoreProperties.PROJECT_NAME_PROPERTY);
```

Comment:
This is somehow breaking backward compatibility. What about logs that were logging each module name? Can't we keep a "non null" behavior, and keep a way to know the original value to populate the scanner report?